iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -196,9 +201,15 @@ public void commitCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTabl
       if (Catalogs.hiveCatalog(conf, catalogProperties)) {
         catalogProperties.put(TableProperties.ENGINE_HIVE_ENABLED, true);
       }
-
-      Catalogs.createTable(conf, catalogProperties);
+      String metadataLocation = hmsTable.getParameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
```

Comment:
Could we store the location in a `HiveIcebegMetaHook` attribute instead of sending it to HMS?

Reply:
I could extend the `HiveMetaHook` interface with an additional `boolean createHMSTableInHook()` that could return `true` if the location property is present. This way the `HiveMetastoreClient` could also evaluate the need of creating the hms table in place or leaving it for the hook. What do you say?